### PR TITLE
:sparkles: Implement ViMuRe model

### LINKED ISSUE #20: 🧪 [vimure v0.1-R] Write test_model.R
- [ ] Choose a sparse tensor package. Probably [tensorr](https://cran.r-project.org/web/packages/tensorr/vignettes/introduction.html)?
- [ ] Write [testthat unit tests](https://testthat.r-lib.org/) that will expect the same functionality we have on the Python version

[(Python version)](https://github.com/latentnetworks/vimure/blob/main/src/python/vimure/test/test_model.py)

### LINKED ISSUE #25: 🔨 [vimure v0.1-R] Implement VimureModel
Full [(Python version)](https://github.com/latentnetworks/vimure/blob/main/src/python/vimure/model.py) for reference.

## Requirements

- [x] Make sure #24 is complete before taking on this task
- [x] Make sure #23 is complete before taking on this task

## Tasks

Functions that need to be implemented in R:
- [ ] [vm.model.py#VimureModel::initialize_priors](https://github.com/latentnetworks/vimure/blob/main/src/python/vimure/model.py#L498)
- [ ] [vm.model.py#VimureModel::update_cache](https://github.com/latentnetworks/vimure/blob/main/src/python/vimure/model.py#L582)
- [ ] [vm.model.py#VimureModel::update_gamma](https://github.com/latentnetworks/vimure/blob/main/src/python/vimure/model.py#L614)
- [ ]  [vm.model.py#VimureModel::update_phi](https://github.com/latentnetworks/vimure/blob/main/src/python/vimure/model.py#L639)
- [ ]  [vm.model.py#VimureModel::update_rho](https://github.com/latentnetworks/vimure/blob/main/src/python/vimure/model.py#L669)
- [ ]  [vm.model.py#VimureModel::update_nu](https://github.com/latentnetworks/vimure/blob/main/src/python/vimure/model.py#L713)

### PR DESCRIPTION
**Pull request by gabiborges1**
Closes #25
Closes #20 
Related to #21  

## Changes
- Add VimureModel binding
- Extend `summary` to summarise vimure model
- Add `build_custom_theta` function
- Add `parse_graph_from_edgelist`, `parse_graph_from_csv`, `parse_graph_from_igraph`
- `vimure` should work with graph and matrix inputs
- Add Rmarkdown

## How to validate
- [x]  Clone this repository
- [x]  Checkout to `25-vimure-v01-r-implement-vimuremodel` and set the working directory to **src/R**
- [x]  Open Rstudio and run:
```{R}
library(devtools)
install()
```

If it is not your first time running vimure on R, R will ask if you want to remove `vimure` from virtual enviroment. Answer Yes. This is necessary to install the developing version of the Python package.

- [x]  Validate the scripts in **notebooks/R**
- [x] Validate the docs help(vimure)
- [x] Typo errors

### CODE REVIEW

**Comment by gabiborges1** [comment]
This PR is ready for review @jonjoncardoso

**Comment by jonjoncardoso** [comment]
> This PR is ready for review @jonjoncardoso

Thanks! Working on it now

File: src/python/vimure/test/test_model.py [hunk 1]
```diff
@@ -165,76 +165,6 @@ def test_vimure_model_extreme_scenarios_over_reporting(self):
 
         assert np.allclose(f1_score(Y_true, Y_rec), 0.92, atol=1e-2)
 
-        def test_vimure_model_extreme_scenarios_over_reporting(self):
```
[the remark this answers is not in the document]
Comment (gabiborges1) [reply src/python/vimure/test/test_model.py:168]: Its duplicated [(source)](https://github.com/latentnetworks/vimure/blob/25-vimure-v01-r-implement-vimuremodel/src/python/vimure/test/test_model.py#L98)

**Review by jonjoncardoso — APPROVED** [review]
I ran all the notebooks and it all works perfectly.

**Pull request merged**